beacon_node/http_api/src/publish_blocks.rs

```diff
@@ -59,28 +86,29 @@ pub async fn publish_block<T: BeaconChainTypes, B: IntoGossipVerifiedBlockConten
 ) -> Result<Response, Rejection> {
     let seen_timestamp = timestamp_now();
 
-    let (block_contents, is_locally_built_block) = match provenanced_block {
-        ProvenancedBlock::Local(block_contents, _) => (block_contents, true),
-        ProvenancedBlock::Builder(block_contents, _) => (block_contents, false),
+    let (unverified_block, unverified_blobs, is_locally_built_block) = match provenanced_block {
+        ProvenancedBlock::Local(block, blobs, _) => (block, blobs, true),
+        ProvenancedBlock::Builder(block, blobs, _) => (block, blobs, false),
     };
     let provenance = if is_locally_built_block {
         "local"
     } else {
         "builder"
     };
-    let block = block_contents.inner_block().clone();
-    let delay = get_block_delay_ms(seen_timestamp, block.message(), &chain.slot_clock);
+    let block = unverified_block.inner_block();
     debug!(log, "Signed block received in HTTP API"; "slot" => block.slot());
     let malicious_withhold_count = chain.config.malicious_withhold_count;
     let chain_cloned = chain.clone();
 
     /* actually publish a block */
-    let publish_block = move |block: Arc<SignedBeaconBlock<T::EthSpec>>,
-                              blobs_opt: Option<BlobSidecarList<T::EthSpec>>,
-                              data_cols_opt: Option<DataColumnSidecarList<T::EthSpec>>,
-                              sender,
-                              log,
-                              seen_timestamp| {
+    let publish_block_p2p = move |block: Arc<SignedBeaconBlock<T::EthSpec>>,
+                                  should_publish: bool,
```

Comment:
might be clearer to rename this to `should_publish_block`

Reply:
michael explained to me offline the blobs should be published with gossip validation level so this is correct, and they *won't* be published when the block gossip validation fails for consensus level validation because this is called in `publish_fn` only after the block + blobs are available

Reply:
yeah, renamed to `should_publish_block` as that's much clearer 0f7b5a14c3fd5bb648434a1d99b99c20229cc58f